### PR TITLE
RavenDB-14779 - Support one way replication from v2.x and v3.x

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-14779

### CODE REVIEW

File: src/Raven.Server/Documents/BlittableMetadataModifier.cs [hunk 1]
```diff
@@ -501,26 +502,51 @@ private unsafe bool AboutToReadPropertyNameInMetadataUnlikely(IJsonParser reader
...
                     {
                         aboutToReadPropertyName = true;
                         return true;
                     }
 
+                    var longValue = *(long*)(state.StringBuffer + sizeof(int) + sizeof(short));
+                    var intValue = *(int*)(state.StringBuffer + sizeof(int) + sizeof(short) + sizeof(long));
+                    if ((longValue != 7237087983830262092 || intValue != 1701406313) &&
+                        (longValue != 5561212665464644932 || intValue != 1701540449))
+                    {
```
Comment (ayende) [inline src/Raven.Server/Documents/BlittableMetadataModifier.cs]: Add comments for the values you are matching on

File: src/Raven.Server/Documents/Handlers/LegacyReplicationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using DatabaseSmuggler = Raven.Server.Smuggler.Documents.DatabaseSmuggler;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class LegacyReplicationHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/replication/lastEtag", "GET", AuthorizationStatus.ValidUser)]
+        public Task LastEtag()
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/LegacyReplicationHandler.cs:26]: This isn't going to be able to run from 2.x / 3.x - there is no way for the 2.x / 3.x to authenticate using client certs.

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Handlers/LegacyReplicationHandler.cs:26]: https://github.com/ravendb/ravendb/pull/10361

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs [hunk 3]
```diff
@@ -672,12 +676,22 @@ private void SetDocumentFlags(DocumentItem item, BuildVersionType buildType)
             }
         }
 
-        private bool SkipDocument(DocumentItem item, BuildVersionType buildType)
+        private bool SkipDocument(DocumentItem item, BuildVersionType buildType, IDocumentActions actions, SmugglerProgressBase.CountsWithLastEtag resultTombstones, out bool isLegacyDelete)
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs]: I do not like that this method is handling delete markers inside. Handle them properly outside. Maybe in TryHandleLegacyDocuments() or something similar.

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs (hunk 3 above)
Comment (grisha-kotler) [reply src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs]: Done

**Review by ppekrol — APPROVED** [review]

**Comment by ppekrol** [comment]
Rebased.

**Pull request merged**